File: Bridge/System/Collections/Concurrent/ConcurrentDictionary.cs
using Bridge;
using System.Collections.Generic;

namespace System.Collections.Concurrent
{
    public class ConcurrentDictionary<TKey, TValue> : IDictionary<TKey, TValue>,
      ICollection<KeyValuePair<TKey, TValue>>, IEnumerable<KeyValuePair<TKey, TValue>>,
      IDictionary, ICollection, IEnumerable
    {
        public ConcurrentDictionary() : this(EqualityComparer<TKey>.Default)
        {
        }

        public ConcurrentDictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection)
            : this(collection, EqualityComparer<TKey>.Default)
        {
        }

        public ConcurrentDictionary(IEqualityComparer<TKey> comparer)
        {
        }

        public ConcurrentDictionary(IEnumerable<KeyValuePair<TKey, TValue>> collection, IEqualityComparer<TKey> comparer)
            : this(comparer)
        {
        }

        public ConcurrentDictionary(int concurrencyLevel, int capacity)
            : this(EqualityComparer<TKey>.Default)
        {
        }

        public ConcurrentDictionary(int concurrencyLevel,
                                     IEnumerable<KeyValuePair<TKey, TValue>> collection,
                                     IEqualityComparer<TKey> comparer)
            : this(collection, comparer)
        {
        }

        // Parameters unused
        public ConcurrentDictionary(int concurrencyLevel, int capacity, IEqualityComparer<TKey> comparer)
            : this(comparer)
        {
        }

        void CheckKey(TKey key)
        {
            throw new NotImplementedException();
        }

        void Add(TKey key, TValue value)
        {
            throw new NotImplementedException();
        }

        void IDictionary<TKey, TValue>.Add(TKey key, TValue value)
        {
            throw new NotImplementedException();
        }

        public bool TryAdd(TKey key, TValue value)
        {
            throw new NotImplementedException();
        }

        void ICollection<KeyValuePair<TKey, TValue>>.Add(KeyValuePair<TKey, TValue> pair)
        {
            throw new NotImplementedException();
        }

        public TValue AddOrUpdate(TKey key, Func<TKey, TValue> addValueFactory, Func<TKey, TValue, TValue> updateValueFactory)
        {
            throw new NotImplementedException();
        }

        public TValue AddOrUpdate(TKey key, TValue addValue, Func<TKey, TValue, TValue> updateValueFactory)
        {
            throw new NotImplementedException();
        }

        TValue AddOrUpdate(TKey key, TValue addValue, TValue updateValue)
        {
            throw new NotImplementedException();
        }

        TValue GetValue(TKey key)
        {
            throw new NotImplementedException();
        }

        public bool TryGetValue(TKey key, out TValue value)
        {
            throw new NotImplementedException();
        }

        public bool TryUpdate(TKey key, TValue newValue, TValue comparisonValue)
        {
            throw new NotImplementedException();
        }

        public TValue this[TKey key]
        {
            get
            {
                throw new NotImplementedException();
            }
            set
            {
                throw new NotImplementedException();
            }
        }

        public TValue GetOrAdd(TKey key, Func<TKey, TValue> valueFactory)
        {
            throw new NotImplementedException();
        }

        public TValue GetOrAdd(TKey key, TValue value)
        {
            throw new NotImplementedException();
        }

        public bool TryRemove(TKey key, out TValue value)
        {
            throw new NotImplementedException();
        }

        bool Remove(TKey key)
        {
            throw new NotImplementedException();
        }

        bool IDictionary<TKey, TValue>.Remove(TKey key)
        {
            throw new NotImplementedException();
        }

        bool ICollection<KeyValuePair<TKey, TValue>>.Remove(KeyValuePair<TKey, TValue> pair)
        {
            throw new NotImplementedException();
        }

        public bool ContainsKey(TKey key)
        {
            throw new NotImplementedException();
        }

        object IDictionary.this[object key]
        {
            get
            {
                throw new NotImplementedException();
            }
            set
            {
                throw new NotImplementedException();
            }
        }

        void IDictionary.Add(object key, object value)
        {
            throw new NotImplementedException();
        }

        bool ICollection<KeyValuePair<TKey, TValue>>.Contains(KeyValuePair<TKey, TValue> pair)
        {
            throw new NotImplementedException();
        }

        public KeyValuePair<TKey, TValue>[] ToArray()
        {
            throw new NotImplementedException();
        }

        public void Clear()
        {
            throw new NotImplementedException();
        }

        public int Count
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public bool IsEmpty
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        bool ICollection<KeyValuePair<TKey, TValue>>.IsReadOnly
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        bool IDictionary.IsReadOnly
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public new ICollection<TKey> Keys
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        public ICollection<TValue> Values
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        ICollection IDictionary.Keys
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        ICollection IDictionary.Values
        {
            get
            {
                throw new NotImplementedException();
            }
        }

        ICollection<T> GetPart<T>(Func<KeyValuePair<TKey, TValue>, T> extractor)
        {
            throw new NotImplementedException();
        }

        void ICollection.CopyTo(Array array, int startIndex)
        {
            throw new NotImplementedException();
        }

        void CopyTo(KeyValuePair<TKey, TValue>[] array, int startIndex)
        {
            throw new NotImplementedException();
        }

        void ICollection<KeyValuePair<TKey, TValue>>.CopyTo(KeyValuePair<TKey, TValue>[] array, int startIndex)
        {
            throw new NotImplementedException();
        }

        void CopyTo(KeyValuePair<TKey, TValue>[] array, int startIndex, int num)
        {
            throw new NotImplementedException();
        }

        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
        {
            throw new NotImplementedException();
        }

        IEnumerator IEnumerable.GetEnumerator()
        {
            throw new NotImplementedException();
        }

        IEnumerator<KeyValuePair<TKey, TValue>> GetEnumeratorInternal()
        {
            throw new NotImplementedException();
        }
        
        class ConcurrentDictionaryEnumerator : IDictionaryEnumerator
        {
            public ConcurrentDictionaryEnumerator(IEnumerator<KeyValuePair<TKey, TValue>> internalEnum)
            {
            }

            public bool MoveNext()
            {
                throw new NotImplementedException();
            }

            public void Reset()
            {
                throw new NotImplementedException();
            }

            public object Current
            {
                get
                {
                    throw new NotImplementedException();
                }
            }

            public DictionaryEntry Entry
            {
                get
                {
                    throw new NotImplementedException();
                }
            }

            public object Key
            {
                get
                {
                    throw new NotImplementedException();
                }
            }

            public object Value
            {
                get
                {
                    throw new NotImplementedException();
                }
            }
        }
        
        static KeyValuePair<U, V> Make<U, V>(U key, V value)
        {
            throw new NotImplementedException();
        }

        uint Hash(TKey key)
        {
            throw new NotImplementedException();
        }

        public bool ContainsKey(object key)
        {
            throw new NotImplementedException();
        }

        public bool Remove(object key)
        {
            throw new NotImplementedException();
        }
    }
}

File: Bridge/Resources/.generated/system/collections/concurrent/concurrentDictionary.js
    Bridge.define("System.Collections.Concurrent.ConcurrentDictionary$2", function (TKey, TValue) { return {
        inherits: [System.Collections.Generic.IDictionary$2(TKey,TValue),System.Collections.Generic.ICollection$1(System.Collections.Generic.KeyValuePair$2(TKey,TValue)),System.Collections.Generic.IEnumerable$1(System.Collections.Generic.KeyValuePair$2(TKey,TValue)),System.Collections.IDictionary,System.Collections.ICollection,System.Collections.IEnumerable],
        statics: {
            methods: {
                make: function (U, V, key, value) {
                    throw new System.NotImplementedException();
                }
            }
        },
        props: {
            Count: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            IsEmpty: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            System$Collections$Generic$ICollection$1$IsReadOnly: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            System$Collections$IDictionary$IsReadOnly: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Keys: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            Values: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            System$Collections$IDictionary$Keys: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            },
            System$Collections$IDictionary$Values: {
                get: function () {
                    throw new System.NotImplementedException();
                }
            }
        },
        alias: [
            "System$Collections$Generic$IDictionary$2$add", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$add",
            "System$Collections$Generic$ICollection$1$add", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$add",
            "tryGetValue", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$tryGetValue",
            "getItem", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$getItem",
            "setItem", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$setItem",
            "System$Collections$Generic$IDictionary$2$remove", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$remove",
            "System$Collections$Generic$ICollection$1$remove", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$remove",
            "containsKey", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$containsKey",
            "System$Collections$Generic$ICollection$1$contains", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$contains",
            "clear", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$clear",
            "Count", "System$Collections$ICollection$Count",
            "Count", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$Count",
            "System$Collections$Generic$ICollection$1$IsReadOnly", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$IsReadOnly",
            "Keys", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$Keys",
            "Values", "System$Collections$Generic$IDictionary$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$Values",
            "System$Collections$Generic$ICollection$1$copyTo", "System$Collections$Generic$ICollection$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$copyTo",
            "getEnumerator", ["System$Collections$Generic$IEnumerable$1$System$Collections$Generic$KeyValuePair$2$" + Bridge.getTypeAlias(TKey) + "$" + Bridge.getTypeAlias(TValue) + "$getEnumerator", "System$Collections$Generic$IEnumerable$1$getEnumerator"],
            "containsKey$1", "System$Collections$IDictionary$containsKey",
            "remove", "System$Collections$IDictionary$remove"
        ],
        ctors: {
            ctor: function () {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor3.call(this, System.Collections.Generic.EqualityComparer$1(TKey).def);
            },
            $ctor1: function (collection) {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor2.call(this, collection, System.Collections.Generic.EqualityComparer$1(TKey).def);
            },
            $ctor3: function (comparer) {
                this.$initialize();
            },
            $ctor2: function (collection, comparer) {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor3.call(this, comparer);
            },
            $ctor5: function (concurrencyLevel, capacity) {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor3.call(this, System.Collections.Generic.EqualityComparer$1(TKey).def);
            },
            $ctor4: function (concurrencyLevel, collection, comparer) {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor2.call(this, collection, comparer);
            },
            $ctor6: function (concurrencyLevel, capacity, comparer) {
                System.Collections.Concurrent.ConcurrentDictionary$2(TKey,TValue).$ctor3.call(this, comparer);
            }
        },
        methods: {
            getItem: function (key) {
                throw new System.NotImplementedException();
            },
            setItem: function (key, value) {
                throw new System.NotImplementedException();
            },
            getItem: function (key) {
                throw new System.NotImplementedException();
            },
            setItem: function (key, value) {
                throw new System.NotImplementedException();
            },
            checkKey: function (key) {
                throw new System.NotImplementedException();
            },
            add: function (key, value) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$IDictionary$2$add: function (key, value) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$ICollection$1$add: function (pair) {
                throw new System.NotImplementedException();
            },
            System$Collections$IDictionary$add: function (key, value) {
                throw new System.NotImplementedException();
            },
            tryAdd: function (key, value) {
                throw new System.NotImplementedException();
            },
            addOrUpdate$1: function (key, addValueFactory, updateValueFactory) {
                throw new System.NotImplementedException();
            },
            addOrUpdate: function (key, addValue, updateValueFactory) {
                throw new System.NotImplementedException();
            },
            addOrUpdate$2: function (key, addValue, updateValue) {
                throw new System.NotImplementedException();
            },
            getValue: function (key) {
                throw new System.NotImplementedException();
            },
            tryGetValue: function (key, value) {
                throw new System.NotImplementedException();
            },
            tryUpdate: function (key, newValue, comparisonValue) {
                throw new System.NotImplementedException();
            },
            getOrAdd$1: function (key, valueFactory) {
                throw new System.NotImplementedException();
            },
            getOrAdd: function (key, value) {
                throw new System.NotImplementedException();
            },
            tryRemove: function (key, value) {
                throw new System.NotImplementedException();
            },
            remove$1: function (key) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$IDictionary$2$remove: function (key) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$ICollection$1$remove: function (pair) {
                throw new System.NotImplementedException();
            },
            remove: function (key) {
                throw new System.NotImplementedException();
            },
            containsKey: function (key) {
                throw new System.NotImplementedException();
            },
            containsKey$1: function (key) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$ICollection$1$contains: function (pair) {
                throw new System.NotImplementedException();
            },
            toArray: function () {
                throw new System.NotImplementedException();
            },
            clear: function () {
                throw new System.NotImplementedException();
            },
            getPart: function (T, extractor) {
                throw new System.NotImplementedException();
            },
            System$Collections$ICollection$copyTo: function (array, startIndex) {
                throw new System.NotImplementedException();
            },
            copyTo: function (array, startIndex) {
                throw new System.NotImplementedException();
            },
            System$Collections$Generic$ICollection$1$copyTo: function (array, startIndex) {
                throw new System.NotImplementedException();
            },
            copyTo$1: function (array, startIndex, num) {
                throw new System.NotImplementedException();
            },
            getEnumerator: function () {
                throw new System.NotImplementedException();
            },
            System$Collections$IEnumerable$getEnumerator: function () {
                throw new System.NotImplementedException();
            },
            getEnumeratorInternal: function () {
                throw new System.NotImplementedException();
            },
            hash: function (key) {
                throw new System.NotImplementedException();
            }
        }
    }; });
